### PR TITLE
Restructuring the API guide

### PR DESCRIPTION
**Pull request by cloveras**
Version 2. Lots and lots of changes, mainly:

- Updated flow diagram (PlantUML and SVG)
- Sections based on each API endpoint
- Added a few screenshots/examples
- Removed numerous duplicated code examples
- Simplified and clarified _en masse_
- Language and typo fixes

### CODE REVIEW

File: vipps-ecom-api.md [hunk 1]
```diff
@@ -1143,26 +1184,30 @@ This may be done in two ways:
...
+After the user has finished (or cancelled) the payment in the Vipps app, the user is returned back to the browser or native app that started the payment flow (via a the fallback URL provided by the merchant when the order is created). The merchant app or website should then query the ecom API for the updated state of the payment operation.
 
-*NOTE:* When the user arrives back in the merchant app or website, we strongly recommend that you perform a call to the ecom api to check the state of the transaction.
-While some of the state of the ecom operation *can* be derived from things like wether or not user returned successfully from the native app, the most reliable
-approach to know the state of the payment flow is always to query the ecom API once you arrive back in the merchant app/website.
+**Please note:** When the user arrives back in the merchant app or website, we
+_strongly_ recommend that you perform a call to the eCom API to check the state
+of the transaction. While some of the state of the eCom operation *can* be
+derived from things like wether or not user returned successfully from the
```
Comment (tht13) [inline vipps-ecom-api.md]: whether*

File: vipps-ecom-api.md [hunk 2]
```diff
@@ -990,12 +989,12 @@ See the [Swagger specification](./) for more details.
 | `429 Too Many Requests` | There is currently a limit of max 200 calls per second  |
 | `500 Server Error`      | An internal Vipps problem.                              |
 
-HTTP requests that are being stopped in the application gateway will result in
-an error JSON object, while requests that are produced from the backend will
-receive an array with a JSON object. Error codes that are produced from the
-application gateway include `401`, `403` and `422`.
+HTTP responses with errors from the application gateway contain one error JSON object.
+Error responses produced from the application gateway include `401`, `403` and `422`.
```
Comment (tht13) [inline vipps-ecom-api.md]: 429 as well

File: images/vipps-ecom-flow-chart.plantuml [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
 
 [*] --> Initiate
-Initiate --> Reserve :T1
-Initiate --> Cancel :T2
-Reserve --> Capture :T3
-Reserve --> Cancel  :T4
-Capture --> Refund  :T5
+Initiate --> Reserve: User confirms
+Reserve --> Capture: Merchant captures
+Reserve --> Cancel: User cancels
```
[the remark this answers is not in the document]
Comment (cloveras) [reply images/vipps-ecom-flow-chart.plantuml]: @Markusdreyer: Can you please add updated screenshots from your webshop?

File: vipps-ecom-api.md [hunk 4]
```diff
@@ -2,389 +2,223 @@
...
+3. The merchant uses the `vipps://` URL to invoke the Vipps app.
+4. Vipps is automatically opened.
+5. The user accepts or rejects the payment request in Vipps.
+6. When the payment process is completed, Vipps redirects to the merchant using the `fallBack` URL.
 
-Initiate payment is used to create a new payment order in Vipps:
+## Initiate payment flow: API calls
 
+Swagger:
 [`POST:/ecomm/v2/payments`](https://vippsas.github.io/vipps-ecom-api/#/Vipps_eCom_API/initiatePaymentV3UsingPOST)
```
[the remark this answers is not in the document]
Comment (cloveras) [reply vipps-ecom-api.md]: Swagger UI did it.

**Comment by cloveras** [comment]
Merging now, as images are easily replaced, and the improvements are worth getting out ASAP.

File: vipps-ecom-api.md [hunk 5]
```diff
@@ -549,47 +551,87 @@ has an additional 5 minutes to complete the payment in Vipps.
...
+            "shippingMethodId": "321abc"
+        }
+    ]
+}
+```
+
+### Transaction update
+
+A callback to the merchant for receiving post-payment information:
+[`POST:/v2/payments/{orderId}`](https://vippsas.github.io/vipps-ecom-api/#/Endpoints_required_by_Vipps_from_the_merchant/transactionUpdateCallbackForRegularPaymentUsingPOST).
```
[the remark this answers is not in the document]
Comment (tht13) [reply vipps-ecom-api.md]: I meant to the name for the url, ie `POST:/v2/payments/{orderId}` -> `POST:[callbackprefix]/v2/payments/{orderId}`

**Pull request merged**